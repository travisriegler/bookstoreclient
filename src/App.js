import axios from 'axios';
import baseUrl from './config'

const App = () => {

  axios(`${baseUrl}/api/v1/books`).then(books => {
    console.log(books);
  })

  return <div>My first component</div>
}

export default App;