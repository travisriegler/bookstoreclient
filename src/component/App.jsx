import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import Layout from './layout/Layout';
import BookContainer from './book/BookContainer';
import Login from './user/Login';
import { SnackbarProvider } from 'notistack'
import Auth from './auth';
import Register from './user/Register';

const App = () => {

  return (
    <SnackbarProvider maxSnack={3}>
      <Router>
          <Layout>
            <Routes>
                <Route path="/login" element={<Login />} />
                <Route path="/register" element={<Register />} />
                <Route path="/" element={
                  <Auth>
                      <BookContainer />
                  </Auth>
                } />
            </Routes>
          </Layout> 
      </Router>
    </SnackbarProvider>
  )
}

export default App;