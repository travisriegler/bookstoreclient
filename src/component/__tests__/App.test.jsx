import React from "react";
import { render } from "@testing-library/react";
import App from "../App";
import renderWithRedux from "../../util/testUtil";

describe("App component", () => {
    it("Should render app without error", () => {
        const { asFragment } = renderWithRedux(<App />, {});
        expect(asFragment()).toMatchSnapshot();
    });
});