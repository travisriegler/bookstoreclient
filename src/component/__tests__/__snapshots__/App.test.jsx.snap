// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Should render app without error 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiBox-root css-0"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-10ltm04-MuiToolbar-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1qp5wg9-MuiTypography-root"
          >
            Book Store
          </h6>
        </div>
      </header>
    </div>
    <div
      class="MuiBox-root css-1afphko"
    >
      <form
        autocomplete="off"
        novalidate=""
      >
        <div
          class="MuiBox-root css-1wcjc1k"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-tjzra3-MuiPaper-root"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
            >
              Book Store Login
            </h4>
            <div
              class="MuiFormControl-root MuiTextField-root css-7xtz12-MuiFormControl-root-MuiTextField-root"
              data-testid="email-testid"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="email"
                id="email-label"
              >
                Enter email address
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="email"
                  name="email"
                  placeholder="Enter email address"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Enter email address
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root css-7xtz12-MuiFormControl-root-MuiTextField-root"
              data-testid="password-testid"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="password"
                id="password-label"
              >
                Enter password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="password"
                  name="password"
                  placeholder="Enter password"
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Enter password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              style="margin-top: 2rem;"
              tabindex="0"
              type="submit"
            >
              Login
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <br />
            <button
              class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways MuiLink-button css-z246bh-MuiTypography-root-MuiLink-root"
            >
              Register
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
